packages/expo-device/android/src/main/java/expo/modules/device/DeviceModule.kt

```diff
@@ -0,0 +1,209 @@
+    "totalMemory" to run {
+      val memoryInfo = ActivityManager.MemoryInfo()
+      (mContext.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager).getMemoryInfo(memoryInfo)
+      memoryInfo.totalMem
+    },
+    "supportedCpuArchitectures" to run {
+      var supportedAbis = Build.SUPPORTED_ABIS
+      if (supportedAbis != null && supportedAbis.isEmpty()) {
+        supportedAbis = null
+      }
+      supportedAbis
+    },
+    "osName" to systemName,
+    "osVersion" to Build.VERSION.RELEASE,
+    "osBuildId" to Build.DISPLAY,
+    "osInternalBuildId" to Build.ID,
+    "osBuildFingerprint" to Build.FINGERPRINT,
+    "platformApiLevel" to Build.VERSION.SDK_INT,
+    "deviceName" to Settings.Secure.getString(mContext.contentResolver, "bluetooth_name")
+  )
+
+  private val deviceYearClass: Int
+    get() = YearClass.get(mContext)
+
+  private val systemName: String
+    get() {
+      return if (Build.VERSION.SDK_INT < 23) {
+        "Android"
+      } else {
+        Build.VERSION.BASE_OS.takeIf { it.isNotEmpty() } ?: "Android"
+      }
+    }
+
+  @ExpoMethod
+  fun getDeviceTypeAsync(promise: Promise) {
+    promise.resolve(getDeviceType(mContext).JSValue)
+  }
+
+  @ExpoMethod
+  fun getUptimeAsync(promise: Promise) {
+    promise.resolve(SystemClock.uptimeMillis().toDouble())
+  }
+
+  @ExpoMethod
+  fun getMaxMemoryAsync(promise: Promise) {
+    val maxMemory = Runtime.getRuntime().maxMemory()
+    promise.resolve(if (maxMemory != Long.MAX_VALUE) maxMemory.toDouble() else -1)
+  }
+
+  @ExpoMethod
+  fun isRootedExperimentalAsync(promise: Promise) {
+    var isRooted = false
+    val isDevice = !isRunningOnGenymotion && !isRunningOnStockEmulator
+
+    try {
+      val buildTags = Build.TAGS
+      isRooted = if (isDevice && buildTags != null && buildTags.contains("test-keys")) {
+        true
+      } else {
+        if (File("/system/app/Superuser.apk").exists()) {
+          true
+        } else {
+          isDevice && File("/system/xbin/su").exists()
+        }
+      }
+    } catch (se: SecurityException) {
+      promise.reject(
+        "ERR_DEVICE_ROOT_DETECTION",
+        "Could not access the file system to determine if the device is rooted.",
+        se
+      )
+      return
+    }
+
+    promise.resolve(isRooted)
+  }
+
+  @ExpoMethod
+  fun isSideLoadingEnabledAsync(promise: Promise) {
+    val enabled: Boolean = if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+      Settings.Global.getInt(
+        mContext.applicationContext.contentResolver,
+        Settings.Global.INSTALL_NON_MARKET_APPS,
+        0
+      ) == 1
+    } else {
+      mContext.applicationContext.packageManager.canRequestPackageInstalls()
+    }
+
+    promise.resolve(enabled)
+  }
+
+  @ExpoMethod
+  fun getPlatformFeaturesAsync(promise: Promise) {
+    val allFeatures = mContext.applicationContext.packageManager.systemAvailableFeatures
+    val featureList = allFeatures.filter { it.name != null }.map { it.name }
```

Comment:
```suggestion
    val featureList = allFeatures.filterNotNull().map { it.name }
```
I think `filterNotNull` should be available.

Reply:
Indeed, it is!